NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -36,7 +36,12 @@ double norm(GlobalVector const& x, unsigned const global_component,
         }
     }
 
-    // TODO for PETSc some global accumulation is necessary.
+#ifdef USE_PETSC
```

Comment:
This could be an inaccurate temporary solution  because the values of the entries of the global x are fetched individually and the values of the ghost nodes are not excluded.

One better solution under PETSc is that to replace the above norm computation code with
1. call  VecGetSubVector(Vec X,IS is,Vec *Y) to get a sub vector for a component, where IS can be set by using DOF table. Or call VecGetValues(Vec x,PetscInt ni,const PetscInt ix[],PetscScalar y[]) to get an array of a component.
2. compute the norm directly.